narwhal/primary/src/primary.rs

```diff
@@ -280,6 +291,10 @@ impl Primary {
                 outbound_network_metrics,
             )))
             .layer(CallbackLayer::new(FailpointsMakeCallbackHandler::new()))
+            .layer(SetRequestHeaderLayer::overriding(
+                EPOCH_HEADER_KEY.parse().unwrap(),
+                epoch_string,
+            ))
```

Comment:
For debugging purposes maybe we could add a SetResponseHeaderLayer setting the epoch header on all responses that narwhal returns as well?